src/ngx_http_lua_directive.c

```diff
@@ -464,8 +466,20 @@ ngx_http_lua_rewrite_by_lua(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
-    if (llcf->rewrite_handler) {
-        return "is duplicate";
+    if (p_rewrites == NULL) {
+        p_rewrites = ngx_array_create(cf->pool, 8,
```

Comment:
Multiple same-phase handlers should be rare. We can set the initial size to 1 instead of 8.